cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -177,9 +177,12 @@ class fixed_width_scalar : public scalar {
   void set_value(T value, rmm::cuda_stream_view stream = cudf::get_default_stream());
 
   /**
-   * @brief Explicit conversion operator to get the value of the scalar on the host.
+   * @brief Returns the value of the scalar on the host.
+   *
+   * @param stream CUDA stream used for device memory operations.
+   * @return The value of the scalar
    */
-  explicit operator value_type() const;
+  T get_value(rmm::cuda_stream_view stream = cudf::get_default_stream()) const;
```

Comment:
RMM just calls this `value`. We often try to avoid “get” in our function names.
```suggestion
  T value(rmm::cuda_stream_view stream = cudf::get_default_stream()) const;
```

Reply:
Wait, doesn’t `value` already exist just below here? We may not need `get_value` at all. Maybe we just need to delete the conversion operator since it does not take a stream.

Reply:
Good point - I chose to introduce another function here because `value` can return either host or device data depending on the type. `string_scalar::value` returns a `cudf::string_view` on the device, while `fixed_point::value` and `fixed_width::value` copy to host. I was also unsure of using `fixed_point::value` directly since the conversion operator returns the `fixed_point_value` instead of the unscaled value.

Reply:
I'm hesitant to modify `string_scalar::value` to return a host string view since it is used in several places that directly access the scalar on device. One option is to rename this function to `string_scalar::d_value` and introduce another `string_scalar::value` that returns a `std::string`?

Reply:
The `string_scalar::value()` should return a `cudf::string_view` which is possible from host or device since it just wraps a pointer and a size held by the `string_scalar`. No device code is needed. Actually the `stream` parameter is not used at all.
To return a host `std::string` one would use the [`string_scalar::to_string()`](https://github.com/rapidsai/cudf/blob/b65ad08be4e10ae4e95baf76b74de9d022331ba8/cpp/include/cudf/scalar/scalar.hpp#L529) function.

Reply:
I agree with Bradley we should not have the `get_value()` functions and just keep the `value()` ones.

Reply:
Thanks David, Bradley. I've retained the `value` functions and got rid of the conversion operators.